Version 2.0.0 (#33)
ScottBrenner/cfn-lint-action: README.md

# GitHub Action for CloudFormation Linter

This Action for [CloudFormation Linter](https://github.com/aws-cloudformation/cfn-python-lint/) enables arbitrary actions for interacting with CloudFormation Linter to validate CloudFormation yaml/json templates against the CloudFormation spec and additional checks. Includes checking valid values for resource properties and best practices.

> The actions primary purpose is to set the [CloudFormation Linter](https://github.com/aws-cloudformation/cfn-python-lint/) to the runners $PATH so you can access the linter throughout the workflow, and use in a way that suits your application. There is a way to run a [CloudFormation Linter](https://github.com/aws-cloudformation/cfn-python-lint/) command in this workflow though if you like.

Reviewer: ```suggestion
The Action's primary purpose is to set the [CloudFormation Linter](https://github.com/aws-cloudformation/cfn-python-lint/) to the runner's $PATH so you can access the linter throughout the workflow, and use in a way that suits your application. There is a way to run a [CloudFormation Linter](https://github.com/aws-cloudformation/cfn-python-lint/) command in this workflow though if you like.
```

There are multiple ways to consume this GitHub Action.

The recommended approach would be to to use this action to add the [CloudFormation Linter](https://github.com/aws-cloudformation/cfn-python-lint/) to the runners $PATH and run commands throughout the rest of the workflow. An example of that can be found below:

```yaml
name: Lint CloudFormation Templates

on: [push]

jobs:
  cloudformation-linter:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Setup Cloud Formation Linter with Latest Version
        uses: scottbrenner/cfn-lint-action@v2

      - name: Print the Cloud Formation Linter Version & run Linter.
        run: |
          cfn-lint --version
          cfn-lint -t ./template.yml
```

Within the `run |` section, you specify the required Cloud Formation Linter commands.

However, if you would rather run the Cloud Cloud Formation Linter commands within this action. That is also possible. An example of that can be found below:

```yaml
name: Lint CloudFormation Templates

on: [push]

jobs:
  cloudformation-linter:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout
        uses: actions/checkout@v2

>       - name: Testing with CFN Lint Version & Command

Reviewer: ```suggestion
      - name: Testing with CFN Lint Command
```

| Input Name | Input Description                                  | Default Value                                             | Required? |
| ---------- | -------------------------------------------------- | --------------------------------------------------------- | --------- |
| version    | Version of CFN PyPi Package                        | Latest Version of CFN PyPi Package                        | false     |
| python     | Python Version                                     | Defaults to `python` on Windows, and `python3` otherwise. | false     |
| command    | Cloud Formation Linter Command to Run Afer Install | N/A                                                       | false     |

This GitHub Action does not directly output any values.

## How to use the Cloud Formation Linter?

See [Cloud Formation Linter Usage](https://github.com/aws-cloudformation/cfn-python-lint#basic-usage) for full usage details.

## Upgrading from Version 1?

> The main difference between Version 1 and Version 2 is the fact Version 2 by default doesn't run any linting commands. The primary purpose of this action is now to provide the underlying [CloudFormation Linter](https://github.com/aws-cloudformation/cfn-python-lint/) package and enable you to run your commands. To do this, follow the steps below:

Reviewer: ```suggestion
The main difference between Version 1 and Version 2 is the fact Version 2 by default doesn't run any linting commands. The primary purpose of this action is now to provide the underlying [CloudFormation Linter](https://github.com/aws-cloudformation/cfn-python-lint/) package and enable you to run your commands. To upgrade from Version 1 to Version 2, follow the steps below:
```

#### Step One

Change the action in your Workflow to be:

```
- name: Setup Cloud Formation Linter with Latest Version
  uses: scottbrenner/cfn-lint-action@v2
```

#### Step Two

Below the above include the following:

```
- name: Print the Cloud Formation Linter Version & run Linter.
  run: |
    cfn-lint --version

>     cfn-lint -t ./**/*.yaml

Reviewer: Would it be possible to make `cfn-lint -t ./**/*.yaml` the default command for the Action, if no command is specified? Doing so might remove the "breaking change", in the situation where someone is currently using this Action as
```
- name: cfn-lint
      uses: docker://ghcr.io/scottbrenner/cfn-lint-action:master
      with:
        args: "**/*.yaml"
```
If it's not too much trouble, if the Action could continue to just run `cfn-lint -t ./**/*.yaml` by default, Workflows should continue to work just fine if I'm not mistaken.

The Action's primary purpose is to set the [CloudFormation Linter](https://github.com/aws-cloudformation/cfn-python-lint/) to the runner's $PATH so you can access the linter throughout the workflow, and use in a way that suits your application. There is a way to run a [CloudFormation Linter](https://github.com/aws-cloudformation/cfn-python-lint/) command in this workflow though if you like.

## Usage

There are multiple ways to consume this GitHub Action.

The recommended approach would be to to use this action to add the [CloudFormation Linter](https://github.com/aws-cloudformation/cfn-python-lint/) to the runners $PATH and run commands throughout the rest of the workflow. An example of that can be found below:

```yaml
name: Lint CloudFormation Templates

on: [push]

jobs:
  cloudformation-linter:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Setup Cloud Formation Linter with Latest Version

>         uses: scottbrenner/cfn-lint-action@v2

Reviewer: ```suggestion
        uses: scottbrenner/cfn-lint-action@v2.0.0
```
(Planning to tag&release this as `v2.0.0`)

Reviewer: Oh yeah totally, if `v2` works like `v2*` this is the right way to do it.

Author: Yup 👍 Agreed! Just done a little digging and I think we are good so if it's cool with you let's keep the readme at `v2` 👍

      - name: Print the Cloud Formation Linter Version & run Linter.
        run: |
          cfn-lint --version
          cfn-lint -t ./template.yml
```

Within the `run |` section, you specify the required Cloud Formation Linter commands.

> However, if you would rather run the Cloud CloudFormation Linter commands within this action. That is also possible. An example of that can be found below:

Reviewer: ```suggestion
However, if you would rather run the CloudFormation Linter commands within this action. That is also possible. An example of that can be found below:
```

> However, if you would rather run the Cloud Cloud Formation Linter commands within this action. That is also possible. An example of that can be found below:

Reviewer: ```suggestion
However, if you would rather run the Cloud CloudFormation Linter commands within this action. That is also possible. An example of that can be found below:
```

```yaml
name: Lint CloudFormation Templates

on: [push]

jobs:
  cloudformation-linter:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Testing with CFN Lint Command

>         uses: ./

Reviewer: ```suggestion
        uses: scottbrenner/cfn-lint-action@v2.0.0
```
This probably needs to be the Action, right? Don't think `./` will work outside of this repository.

Author: BOOM 👍 correct! See my comment above about the v2 thing @ScottBrenner but I agree this should be: `scottbrenner/cfn-lint-action@*` yes! Thanks for the good find 👍

        with:
          command: cfn-lint -t ./template.yml
```

Further, you can configure this action to download a specific version of the [CloudFormation Linter](https://github.com/aws-cloudformation/cfn-python-lint/), as well as the Python interpreter. See the table below for all the `INPUTS` this action can take.

| Input Name | Input Description                                  | Default Value                                             | Required? |
| ---------- | -------------------------------------------------- | --------------------------------------------------------- | --------- |
| version    | Version of CFN PyPi Package                        | Latest Version of CFN PyPi Package                        | false     |
| python     | Python Version                                     | Defaults to `python` on Windows, and `python3` otherwise. | false     |
| command    | Cloud Formation Linter Command to Run Afer Install | N/A                                                       | false     |

This GitHub Action does not directly output any values.

> ## How to use the Cloud Formation Linter?

Reviewer: ```suggestion
## How to use the CloudFormation Linter?
```

See [Cloud Formation Linter Usage](https://github.com/aws-cloudformation/cfn-python-lint#basic-usage) for full usage details.

## Upgrading from Version 1?

> The main difference between Version 1 and Version 2 is the fact Version 2 by default doesn't run any linting commands. The primary purpose of this action is now to provide the underlying [CloudFormation Linter](https://github.com/aws-cloudformation/cfn-python-lint/) package and enable you to run your commands. To upgrade from Version 1 to Version 2, follow the steps below:

Reviewer: ```suggestion
The main difference between Version 1 and Version 2 is the fact that Version 2 by default doesn't run any linting commands. The primary purpose of this action is now to provide the underlying [CloudFormation Linter](https://github.com/aws-cloudformation/cfn-python-lint/) package and enable you to run your commands. To upgrade from Version 1 to Version 2, follow the steps below:
```

#### Step One

Change the action in your Workflow to be:

```
- name: Setup Cloud Formation Linter with Latest Version

>   uses: scottbrenner/cfn-lint-action@v2

Reviewer: ```suggestion
  uses: scottbrenner/cfn-lint-action@v2.0.0
```

Author: See comment above first 👍

```

#### Step Two

Below the above include the following:

```
- name: Print the Cloud Formation Linter Version & run Linter.
  run: |
    cfn-lint --version
    cfn-lint -t ./**/*.yaml
```

This should give you the same experience as before.

## Running Locally

> This is a JavaScript GitHub Action that relies on Node for third party packages. To get this working on your local machine, firstly make sure you have a working [NodeJS](https://nodejs.org/en/) runtime. It is also recommended to use [Act](https://github.com/nektos/act) to test &amp; run actions locally. These instructions assume you have the [GitHub CLI](https://cli.github.com/) installed.

Reviewer: ```suggestion
This is a JavaScript GitHub Action that relies on Node for third party packages. To get this working on your local machine, firstly make sure you have a working [NodeJS](https://nodejs.org/en/) runtime. It is also recommended to use [Act](https://github.com/nektos/act) to test & run actions locally. These instructions assume you have the [GitHub CLI](https://cli.github.com/) installed.
```
